fluent-plugin-events/lib/sumologic/kubernetes/connector.rb

```diff
@@ -7,17 +7,41 @@ module Connector
       K8_POD_CA_CERT = 'ca.crt'.freeze
       K8_POD_TOKEN = 'token'.freeze
 
-      def connect_kubernetes
-        @client = Kubeclient::Client.new(
-          @kubernetes_url, @api_version,
+      # Need different clients to access different API groups/versions
+      CORE_API_VERSIONS = ['v1'].freeze
+      API_GROUPS = ['events.k8s.io/v1beta1'].freeze
```

Comment:
I believe since we want to support watching changes to other resource types, we would want to include the other API groups like `apps/v1` and `extensions/v1beta1` in order to watch those. For instance, the `deployment` watch endpoint is `/apis/apps/v1/watch/deployments` according to the docs [here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#watch-list-all-namespaces-32).

Reply:
That is a good point. Actually we don't need to create the clients for all of the available API groups here, but just the one that user wants to get events from. For example, if user specifies the config as:
```
    <source>
      @type events
      resource_name deployments
      api_version apps/v1
    </source>
```
then we put `apps/v1` in `API_GROUPS`. If it is `api_version events.k8s.io/v1beta1` then we put `events.k8s.io/v1beta1` in `API_GROUPS`.

We should however always build the core v1 client since we use it for config maps

Reply:
Aha, this makes sense, I forgot we had exposed a config param for this. In that case should we only create the core v1 client, and the client specified by the `api_version` config param in `connector.rb` ?

Reply:
Oh, then I think we don't need to have this `API_GROUPS` then.